### DISCUSSION TITLE
Port layout based on size

### DISCUSSION THREAD

**Discussion by grafandreas** [clientIO/joint#1991] [Q&A]
Dear all,

I am trying to calculate port position based on port size by providing
groups : {
  in : {
    position : inputPortPosition

However, it seems that in ports.mjs l.48 ff the information passed to the functions is reduced to the port's position.
Is there a way to calculate port position based on full port information (i.e. size)?

Thanks,

Andreas

**Comment by grafandreas** [comment]
Just note that there is a code pen, with a different layout approach for ports. Will check that.

**Reply by kumilingus** [reply]
Right, you can use:
**A.** a custom port layout - https://codepen.io/jointjs/pen/yLKGgaR
**B.** port layout args (`dx` and `dy` args)
```js
el.prop('ports/groups/in/position', { name: 'left', args: { dx: -10 }})
```
**C.** change anchor of the port so you don't have to take the size into account. e.g. change `<circle r="10">` to `<circle r="10" cx="-10">` for the `left` port layout.
